src/mobot_client/payments/client.py

```diff
@@ -34,47 +46,68 @@ def _get_default_account_info(self) -> (str, str):
         public_address = account_obj["main_address"]
         return public_address, account_id
 
+    def check_receiver_receipt_status(self, address, receipt):
+        try:
+            return super().check_receiver_receipt_status(address, receipt)
+        except Exception as e:
+            self.logger.exception("Exception raised when checking receipt")
```

Comment:
Isn't something like this already logged by the exception itself (on the next line)? I noticed a lot of similar cases in this file in this commit.

Reply:
The point of this is to catch and log the initial exception, which is not specific, and then replace it with our own for matching purposes.

Reply:
This is also kinda important because the code is still newly async - meaning, there's not a lot of actually checking or logging out of the results of the futures yet. If you just throw 'em in the pool and don't make sure to log before raising, you won't see the traceback.

Reply:
So, this is because of the future not being explicitly checked.